### PR TITLE
update IMGUI blog post with embedded screencast

### PR DESCRIPTION
**Pull request by solvingj**
update IMGUI blog post with embedded screencast, cosmetic improvements, last_modified_at field, and meta fields for SEO. 
We should merge the other PR's i've submitted for SEO and `last_modified_at` field supports before merging this.
We MUST set the youtube video to public access before merging this.

### CODE REVIEW

**Comment by uilianries** [comment]
The screencast is still private on YT. It there any date for releasing publicly?

**Comment by solvingj** [comment]
I just set it to public. Plan to publish/announce shortly.

File: _posts/2019-06-26-An-introduction-to-the-Dear-ImGui-library.markdown [hunk 1]
```diff
@@ -2,239 +2,300 @@
...
-system and platform we have to introduce some binding for our rendering system. Fortunately, there are many premade bindings in Dear
-ImGui's repo. As we will use Dear ImGui v1.69 these are the ones we will need:
+The typical use of ImGui is when you already have a 3D-pipeline enabled
+application like a content creation or game development tool where you want to
+add a GUI. Let's see how easy it is to integrate ImGui in our application. Our
+example application renders a triangle using OpenGL3. We will use GLFW to manage
+window creation and events handling. As ImGui is independent of the rendering
+system and platform we have to introduce some binding for our rendering system.
+Fortunately, there are many premade bindings in Dear ImGui's repo. As we will
+use Dear ImGui v1.69 these are the ones we will need:
```
Comment (czoido) [inline _posts/2019-06-26-An-introduction-to-the-Dear-ImGui-library.markdown:75]: I think we should also update the versions for imgui to the ones of c3i in the blogpost, it's already updated in the examples repo: https://github.com/conan-io/examples/blob/master/libraries/dear-imgui/basic/conanfile.txt
```suggestion
use Dear ImGui v1.74 these are the ones we will need:
```

File: _posts/2019-06-26-An-introduction-to-the-Dear-ImGui-library.markdown [hunk 2]
```diff
@@ -2,239 +2,300 @@
...
 
 * [imgui_impl_opengl3.cpp](https://github.com/ocornut/imgui/blob/v1.69/examples/imgui_impl_opengl3.cpp)
 * [imgui_impl_opengl3.h](https://github.com/ocornut/imgui/blob/v1.69/examples/imgui_impl_opengl3.h)
 * [imgui_impl_glfw.cpp](https://github.com/ocornut/imgui/blob/v1.69/examples/imgui_impl_glfw.cpp)
 * [imgui_impl_glfw.h](https://github.com/ocornut/imgui/blob/v1.69/examples/imgui_impl_glfw.h)
```
Comment (czoido) [inline _posts/2019-06-26-An-introduction-to-the-Dear-ImGui-library.markdown:81]:
```suggestion
* [imgui_impl_opengl3.cpp](https://github.com/ocornut/imgui/blob/v1.74/examples/imgui_impl_opengl3.cpp)
* [imgui_impl_opengl3.h](https://github.com/ocornut/imgui/blob/v1.74/examples/imgui_impl_opengl3.h)
* [imgui_impl_glfw.cpp](https://github.com/ocornut/imgui/blob/v1.74/examples/imgui_impl_glfw.cpp)
* [imgui_impl_glfw.h](https://github.com/ocornut/imgui/blob/v1.74/examples/imgui_impl_glfw.h)

```

File: _posts/2019-06-26-An-introduction-to-the-Dear-ImGui-library.markdown [hunk 3]
```diff
@@ -2,239 +2,300 @@
...
+to handle OpenGL functions loading. We will use ``cmake_multi`` to generate
+projects for Debug and Release configurations. An imports section was also added
+to download the required bindings for GLFW and OpenGL3.
+
+    {% highlight text %}
+    [requires]
+    imgui/1.69@bincrafters/stable
+    glfw/3.2.1@bincrafters/stable
+    glew/2.1.0@bincrafters/stable
+
```
Comment (czoido) [inline _posts/2019-06-26-An-introduction-to-the-Dear-ImGui-library.markdown:268]:
```suggestion
    imgui/1.74
    glfw/3.3.2
    glew/2.1.0
```

File: _posts/2019-06-26-An-introduction-to-the-Dear-ImGui-library.markdown (hunk 1 above)
Comment (solvingj) [reply _posts/2019-06-26-An-introduction-to-the-Dear-ImGui-library.markdown:75]: Really good suggestion, lets do it.

**Comment by solvingj** [comment]
I wonder if I should re-record the video.  Probably not worth it, but now it's going to be awkward.  I guess updating the dependencies should have been the first thing we decided about.  Oh well.

**Comment by czoido** [comment]
> I wonder if I should re-record the video. Probably not worth it, but that's pretty awkward that the versions of things shown in the video will be quite a lot different than the post and example. I guess updating the dependencies should have been the first thing we decided about. Oh well, lesson learned for the next one.

Luckily the video has the good ones 😀  https://youtu.be/O2E-W9P-jKc?t=287

**Comment by solvingj** [comment]
Oh, wow, that is super lucky! Or, I guess, you did it right from the start by updating the examples repo and then the video is based on that... so not lucky at all.  Good work :)

**Comment by uilianries** [comment]
I would update YT video title from "Introduction to the ImGui C++ Library" to "Introduction to the ImGui C++ Library with Conan", to improve search engine result visibility.

**Pull request merged**